desc/equilibrium/equilibrium.py

```diff
@@ -266,8 +283,28 @@ def __init__(
         self._M_grid = setdefault(M_grid, 2 * self.M)
         self._N_grid = setdefault(N_grid, 2 * self.N)
 
-        self._surface.change_resolution(self.L, self.M, self.N, sym=self.sym)
-        self._axis.change_resolution(self.N, sym=self.sym)
+        self._Nz = int(setdefault(Nz, self.surface._Nz))
+        self._Mz = int(setdefault(Mz, self.surface._Mz))
+        self._Lz = int(
+            setdefault(
+                Lz,
+                self.Mz if (self.spectral_indexing == "ansi") else 2 * self.Mz,
+            )
+        )
+
+        # RG: NFP, sym, Mz, Nz passed as kwargs, order is sacrosanct
```

Comment:
just use kwargs for everything if we are adding new arguments here, if that makes it clearer. I am in favor of kwargs over args always